### ISSUE TITLE
Turbo frame request from subdomain after hard reload without cookie header

### ISSUE THREAD

**Issue by mdrbohlav** [hotwired/turbo#1323]
Hi guys, I have an issue with loading turbo frame from subdomain after hard reload (`cmd/ctrl+shift+r` or disabled cache in the Dev console > Network tab). So far I know it happens in Chrome and Safari, everything works in Firefox.

The overview of the setup:
1) Main app runs on root domain, let's say `example.com`. The app there has a turbo frame that returns either user menu if signed in or guest menu if not. The url for that is `https://example.com/-/user_menu`. Everything on this domain works as expected.
2) I have an admin part of the app that runs on a subdomain `admin.example.com`. It has the same user menu so the turbo frame there has the same url `https://example.com/-/user_menu`.

The user menu is loaded correctly when I come to the admin app via a link from the main app. It also works when I reload the page (`cmd/ctrl+r`). But when I do hard reload (`cmd/ctrl+shift+r`) the turbo frame request does not contain the cookie. Could there be an issue with how some browsers handle the cache reset on hard reload and the asynchronous turbo frame load? All the turbo frames loaded later have the cookie set in the request. Lazy loading does not help since this part of the page is at the top.

The user cookie is set up to work with subdomains and the `fetch` has `withCredentials: true`.
  - `domain: .example.com`
  - `same-site: none`
  - `http_only: true`
  - `secure: true`
  - `expiration: in 2025`

Do you have any idea what might be going wrong and where to look for a possible cause?

Ruby version: 3.3.2
Rails version: 7.0.4
Turbo version: 8.0.10
Chrome version: 129.0.6668.90

**Comment by pedroaugustofsilva** [comment]
Hi!

When you say "the fetch has withCredentials: true", what do you mean? Because "withCredentials" is not a valid option for fetch. The correct option is [credentials](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials) and Turbo sets it to "same-origin".

Have you checked the console for any errors?

Let me see if I got this straight. If you access `admin.example.com` in incognito mode, the first load will not send the cookie using src defined in tubo frame. But, in subsequent loads it works?

**Comment by mdrbohlav** [comment]
You are right of course. Do not know where I saw XMLHttpRequest, edited the first post so it does not confuse anyone else. Anyway, that does not change my issue since I have the domain in the cookies set to `.example.com`.

**Comment by vojtad** [comment]
I am following up on @mdrbohlav's report (same org). We are still seeing this issue.

There are no errors in the browser console. The first request doesn't include credentials (`Cookie` header). Subsequent requests include the header.

I don't see any difference in broken and correct requests. They both issue OPTIONS and GET request because they are cross domain requests (from admin.example.com to example.com).

> Let me see if I got this straight. If you access admin.example.com in incognito mode, the first load will not send the cookie using src defined in tubo frame. But, in subsequent loads it works?

That's correct. First load in incognito windows doesn't send cookies to src defined in turbo frame. Subsequent reloads done by Cmd+R, entering the URL again or hitting browser refresh buttons send cookies.

If I use Cmd+Shift+R to reload the page then cookies are not sent again.

We are using following Stimulus controller as a workaround for now:

```javascript
import ApplicationController from 'modules/application_controller';

const MAX_RELOAD_COUNT = 3;

export default class extends ApplicationController {
  #reloadCounter = 0;

  initialize() {
    this.reload = this.#reload.bind(this);
  }

  connect() {
    this.#reloadCounter = 0;
    this.element.addEventListener('turbo:frame-missing', this.reload);
  }

  disconnect() {
    this.element.removeEventListener('turbo:frame-missing', this.reload);
  }

  #reload(event) {
    this.#reloadCounter += 1;

    if (this.#reloadCounter >= MAX_RELOAD_COUNT) {
      console.warn(`Reloading turbo frame ${this.element.id} stopped because max reload count has been reached.`);
      return;
    }

    console.warn(`Reloading turbo frame ${this.element.id} because of missing content.`);

    event.preventDefault();
    setTimeout(() => this.element.reload(), 0);
  }
}
```
